packages/medusa/src/api/routes/admin/products/transaction/create-product-variant.ts

```diff
@@ -78,52 +78,76 @@ export const createVariantTransaction = async (
 
   const productVariantServiceTx = productVariantService.withTransaction(manager)
 
-  async function createVariant(variantInput: CreateProductVariantInput) {
+  async function createVariants(variantInput: CreateProductVariantInput[]) {
     return await productVariantServiceTx.create(productId, variantInput)
   }
 
-  async function removeVariant(variant: ProductVariant) {
-    if (variant) {
-      await productVariantServiceTx.delete(variant.id)
+  async function removeVariants(variants: ProductVariant[]) {
+    if (variants.length) {
+      await productVariantServiceTx.delete(variants.map((v) => v.id))
     }
   }
 
-  async function createInventoryItem(variant: ProductVariant) {
-    if (!variant.manage_inventory) {
-      return
+  async function createInventoryItems(variants: ProductVariant[] = []) {
+    const context = { transactionManager: manager }
+
+    let results: InventoryItemDTO[] = []
+
+    for (const variant of variants) {
```

Comment:
question: is there a reason not to do something like `Promise.all` here?

Reply:
Just wanted to keep it like it was (sequentially) to prevent colliding with you

Reply:
done